modules/@apostrophecms/schema/index.js

```diff
@@ -1614,8 +1614,23 @@ module.exports = {
         } catch (error) {
           throw self.apos.error('invalid', error.message);
         }
-      }
+      },
+
+      stringifyFieldsPatterns(schema) {
```

Comment:
Since we can pass to the frontend a regex type, we simply pass the source (a string) and the frontend will re generate the regex based on this.

Reply:
Schemas are created and passed around in many places. So I suggest moving this to a `validate` function of `string` fields in `addFieldTypes.js`. They don't have a validate function right now, but other field types do, so you can find examples. This will take care of the recursion for you, simplifying your code and making it compatible with more places where schemas are used.

This does mean constructing the regular expressions when needed server side too at convert time, but this isn't a significant performance issue in the context of async code.

Reply:
I'll add a validate method to the string field type to verify the pattern is a regex.
But this method had another purpose, I can't pass a regex to the frontend, I need to mutate it in a string to pass it to the client, that is the purpose of this method.
And for the user experience I want this pattern field property to be a regex and not a string because it feels more natural and allow linter to yell if it's not valid.